apps/client-ui/scripts/health-check-preview.sh

```
@@ -0,0 +1,314 @@
+    
+    print_info "Checking API connectivity..."
+    
+    # Try to extract API URL from the preview environment
+    # This would typically be done by checking the deployed app's configuration
+    # For now, we'll make a basic assumption about the API endpoint
+    
+    local api_url=""
+    
+    # Try to get environment variables from Amplify app
+    if [[ -n "$APP_ID" ]]; then
+        local app_info=$(aws amplify get-app --app-id "$APP_ID" --output json 2>/dev/null || echo "{}")
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)